src/netspresso_trainer/pipelines/detection.py

```diff
@@ -102,18 +117,77 @@ def test_step(self, batch):
         return results
 
     def get_metric_with_all_outputs(self, outputs):
-        targets = np.empty((0, 4))
-        preds = np.empty((0, 5))  # with confidence score
-        targets_indices = np.empty(0)
-        preds_indices = np.empty(0)
+        pred = []
+        targets = []
         for output_batch in outputs:
             for detection, class_idx in output_batch['target']:
-                targets = np.vstack([targets, detection])
-                targets_indices = np.append(targets_indices, class_idx)
+                target_on_image = {}
+                target_on_image['boxes'] = detection
+                target_on_image['labels'] = class_idx
+                targets.append(target_on_image)
 
             for detection, class_idx in output_batch['pred']:
-                preds = np.vstack([preds, detection])
-                preds_indices = np.append(preds_indices, class_idx)
-
-        pred_bbox, pred_confidence = preds[..., :4], preds[..., -1]  # (N x 4), (N,)
-        self.metric_factory.calc((pred_bbox, preds_indices, pred_confidence), (targets, targets_indices), phase='valid')
+                pred_on_image = {}
+                pred_on_image['post_boxes'] = detection[..., :4]
+                pred_on_image['post_scores'] = detection[..., -1]
+                pred_on_image['post_labels'] = class_idx
+                pred.append(pred_on_image)
+        self.metric_factory(pred, target=targets, phase='valid')
```

Comment:
@illian01 

I updated to call with `calc`, instead of `__call__`. Originally, it called with the builtin `__call__` as you modified, but in later it is hard to catch whether the `metric_factory` is method of the class or attribute of the class.

I added `calc` in later, but leave `__call__` just for backward support.

Reply:
@deepkyu 
I got it. I will fix to call `self.metric_factory.calc` method.